nntrainer/layers/conv2d_layer.cpp

```diff
@@ -77,8 +77,8 @@ int Conv2DLayer::initialize(Manager &manager) {
 
   if (weights.empty()) {
     weights.reserve(2);
-    weights.emplace_back(dim, weight_initializer, true, kernelPrefix);
-    weights.emplace_back(bias_dim, bias_initializer, true, biasPrefix);
+    weights.emplace_back(dim, weight_initializer, true, false, kernelPrefix);
+    weights.emplace_back(bias_dim, bias_initializer, true, false, biasPrefix);
```

Comment:
So.. when we initialize layer, there is no memory allocation?

Reply:
Right.

Reply:
Not for the PR, but there is an idea. 

manager calls layer::get_weight() before layer::initialize() and it tracks all the weights. we can extract this to a manager (idea for api change)

Reply:
Sounds good, but this would have to be an added interface and not a replacement for the current method.
With just that interface, a layer, not part of the model would not be able to register weights with manager.

Reply:
If we make sure neuralnetwork calls sth like `manager.trackWeights(layer.getWeights)`. For API point of view, layer does not need to know about the manager.

Reply:
As the API exposes layer functionality, a user cannot then create just a layer (without wrapping it in model), without exposing the manager to the API (which IMO should not be exposed).